paymentsheet-example/src/main/java/com/stripe/android/paymentsheet/example/playground/LinkControllerPlaygroundViewModel.kt

```diff
@@ -1,39 +1,101 @@
 package com.stripe.android.paymentsheet.example.playground
 
 import android.app.Application
+import android.util.Patterns
+import androidx.activity.ComponentActivity
 import androidx.lifecycle.AndroidViewModel
+import androidx.lifecycle.SavedStateHandle
+import androidx.lifecycle.lifecycleScope
+import androidx.lifecycle.viewModelScope
 import com.stripe.android.link.LinkController
 import kotlinx.coroutines.flow.MutableStateFlow
 import kotlinx.coroutines.flow.update
+import kotlinx.coroutines.launch
 
 internal class LinkControllerPlaygroundViewModel(
     application: Application,
+    savedStateHandle: SavedStateHandle,
 ) : AndroidViewModel(application) {
 
+    private val linkController = LinkController.create(application, savedStateHandle)
+    private var linkControllerPresenter: LinkController.Presenter? = null
```

Comment:
alternatively the presenter can live in the activity, and the viewModel could send actions to it via a flow. But this approach sounds good to me. as well! it'd be the integrator's choice.